dynamiqs/mesolve/rouchon.py

```diff
@@ -24,12 +24,42 @@ class MERouchon1(MERouchon):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-        # define cached operators
-        self.M0 = cache(lambda H_nh: self.I - 1j * self.dt * H_nh)  # (b_H, 1, n, n)
-        self.M0dag = cache(lambda M0: M0.mH)  # (b_H, 1, n, n)
-        self.M0rev = cache(lambda H_nh: self.I + 1j * self.dt * H_nh)  # (b_H, 1, n, n)
+        if not self.options.trace_normalization:
+            # define cached operators
+            # self.M0, self.M0rev: (b_H, 1, n, n)
+            # self.M1s: (1, len(jump_ops), n, n)
+            self.M0 = cache(lambda H_nh: self.I - 1j * self.dt * H_nh)
+            self.M0rev = cache(lambda H_nh: self.I + 1j * self.dt * H_nh)
+
+            # define M1s and M1srev
+            self.M1s = sqrt(self.dt) * self.jump_ops
+            self.M1srev = self.M1s
+        else:
+            # compute the inverse square root renormalization matrix
+            # (for time-dependent Hamiltonians, exclude the Hamiltonian from
+            # normalization)
+            # H_nh, M0, M0rev, self.S, self.Srev: (b_H, 1, n, n)
+            if isinstance(self.H, ConstantTDTensor):
+                H_nh_const = self.H_nh(0.0)
+            elif isinstance(self.H, CallableTDTensor):
+                H_nh_const = -0.5j * self.sum_no_jump
+
+            M0 = self.I - 1j * self.dt * H_nh_const
+            M0rev = self.I + 1j * self.dt * H_nh_const
+            self.S = inv_sqrtm(M0.mH @ M0 + self.dt * self.sum_no_jump)
+            self.Srev = inv_sqrtm(M0rev.mH @ M0rev - self.dt * self.sum_no_jump)
+
+            # define cached operators
+            # self.M0, self.M0rev: (b_H, 1, n, n)
+            # self.M1s, self.M1srev: (1, len(jump_ops), n, n)
+            self.M0 = cache(lambda H_nh: self.S - 1j * self.dt * H_nh @ self.S)
+            self.M0rev = cache(lambda H_nh: self.Srev + 1j * self.dt * H_nh @ self.Srev)
+
+            # define M1s and M1srev
+            self.M1s = sqrt(self.dt) * self.S @ self.jump_ops
+            self.M1srev = sqrt(self.dt) * self.Srev @ self.jump_ops
```

Comment:
```suggestion
            self.M1s = sqrt(self.dt) * self.jump_ops @ self.S
            self.M1srev = sqrt(self.dt) * self.jump_ops @ self.Srev
```
Sorry there was a mistake in the previous formula. The S should be on the outside when taking the product \sum_k M_k^\dagger M_k